chain/ethereum/src/network_indexer/ethereum.graphql

```diff
@@ -0,0 +1,83 @@
+""" Block is an Ethereum block."""
+type Block @entity {
+  id: ID!
```

Comment:
Before we start rolling this out, we should brush up the work I did on making it possible to make `ID` equivalent to `Bytes` for a subgraph so that these id's take only 20 instead of 40 (or 42) bytes to store. It will be transparent to the rest of the code, but requires that we pass a flag to `create_subgraph` that indicates whether `ID` should be a `String` or `Bytes`.

Reply:
The code for this is in `store::postgres::relational::Layout::new`, but it's not exposed to callers, and instead capped at `IdType::String` - we should expose this up in the callstack as arguments so it can be set in `create_subgraph` and is stored in the database (maybe as a field on `deployment_schemas`)

Reply:
How much is there to do? And how much, if any, risk does that support introduce? Does it affect clients, filters, anything?

Reply:
Besides passing `IdType::Bytes` or `IdType::String` through when creating a subgraph, the following needs to be done in the `Store`:

* fix up a handful of places in `relational_queries.rs` where we assume that the id is a `String`
* Look at the type of the id column in `information_schema.columns` when starting a subgraph and decide whether it uses `Bytes` or `String` as the id

At the `Entity` layer, the block explorer would have to make sure to pass `id` as a `Value::Bytes` rather than a `Value::String`.

At the GraphQL layer, users have to pass the `id` as something that can be converted to `Value::Bytes`, and we'd have to do that conversion when coercing values. (I could make it so that we convert a `Value::String` into a `Value::Bytes` in `relational_queries.rs`, but that seems a bit hacky)

Reply:
I looked a little more, and to avoid changing too much in the code base, I think the best course of action is to keep that distinction within the relational mapping code. That means that code that deals with entity ID's continues to use strings, and the conversion from string to `bytea` and vice versa all happens in `relational_queries`. For users of the storage layer, the main change is that they might get a new error when the id is not a string in the form `0xdeadbeef`.

Those changes should be possible to do in a couple of days. The only change for the block explorer would be to pass `IdType::Bytes` when creating the schema.

Reply:
Can we do this separately? We're not going to activate this feature right away.

Reply:
Yes, totally; we just need to do this before the feature goes live. We can migrate the database after the fact, but it's likely to take long (maybe hours) and during that time, the block explorer would be unavailable.

When this goes into a release, we need to make sure it's behind a feature flag so that users who install that release don't wind up with this data in their database which we would have to migrate.

Reply:
Opened #1414 to track this properly.

Reply:
👍

Reply:
It's already behind a `--network-subgraphs` CLI flag.